src/ReorderEligibility/ReorderItemPricesEligibilityChecker.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sylius\CustomerReorderPlugin\ReorderEligibility;
+
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Core\Model\OrderItemInterface;
+
+final class ReorderItemPricesEligibilityChecker implements ReorderEligibilityChecker
+{
+    /** @var ReorderEligibilityConstraintMessageFormatterInterface */
+    private $reorderEligibilityConstraintMessageFormatter;
+
+    public function __construct(
+        ReorderEligibilityConstraintMessageFormatterInterface $reorderEligibilityConstraintMessageFormatter
+    ) {
+        $this->reorderEligibilityConstraintMessageFormatter = $reorderEligibilityConstraintMessageFormatter;
+    }
+
+    public function check(OrderInterface $order, OrderInterface $reorder): array
+    {
+        $orderVariantNamesToTotal = [];
+        $reorderVariantNamesToTotal = [];
+
+        /** @var OrderItemInterface $orderItem */
+        foreach ($order->getItems()->getValues() as $orderItem) {
+            $orderVariantNamesToTotal[$orderItem->getVariantName()] = $orderItem->getUnitPrice();
+        }
+
+        /** @var OrderItemInterface $reorderItem */
+        foreach ($reorder->getItems()->getValues() as $reorderItem) {
+            $reorderVariantNamesToTotal[$reorderItem->getVariantName()] = $reorderItem->getUnitPrice();
+        }
+
+        $orderVariantsWithChangedPrice = [];
+
+        foreach (array_keys($orderVariantNamesToTotal) as $variantName) {
+            if (!array_key_exists($variantName, $reorderVariantNamesToTotal)) {
```

Comment:
`!isset($variantName, $reorderVariantNamesToTotal)`

Reply:
I want to check if the key exists in the array, not if the value for given key is set.